crates/full-node/sov-db/src/storage_manager/nomt_based/groups.rs

```diff
@@ -5,23 +5,123 @@ use std::thread::JoinHandle;
 
 use anyhow::Context;
 use rockbound::cache::delta_reader::DeltaReader;
-use rockbound::SchemaBatch;
+use rockbound::rocksdb::ColumnFamilyDescriptor;
+use rockbound::versioned_db::{VersionedDB, VersionedDeltaReader, VersionedTableMetadataKey};
+use rockbound::{default_cf_descriptor, SchemaBatch};
 use sov_rollup_interface::reexports::digest;
 
 use crate::accessory_db::AccessoryDb;
 use crate::config::RollupDbConfig;
-use crate::historical_state::HistoricalStateReader;
+use crate::historical_state::{HistoricalStateReader, StateChanges};
 use crate::ledger_db::LedgerDb;
+use crate::metrics::nomt::PrunerMetric;
 use crate::namespaces::{KernelNamespace, UserNamespace};
 use crate::pruner::Pruner;
-use crate::schema::namespace::StateValues;
-use crate::schema::tables::ModuleAccessoryState;
+use crate::schema::namespace::{
+    NomtCommittedVersion, NomtHistoricalState, NomtPruningState, NomtStateValues,
+};
+use crate::schema::tables::{ModuleAccessoryState, StateRootHashes};
 use crate::state_db_nomt::{NomtSessionBuilder, NomtStateDb, StateOverlay};
 use crate::storage_manager::{update_ledger_finalized_height, InitializableNativeNomtStorage};
+use crate::DbOptions;
+
+/// A database to store the flat state of the rollup (i.e. the raw key-value pairs)
+pub struct FlatStateDb {
+    user: VersionedDB<NomtStateValues<UserNamespace>>,
+    kernel: VersionedDB<NomtStateValues<KernelNamespace>>,
+    other: Arc<rockbound::DB>,
+}
+
+impl FlatStateDb {
+    const DB_NAME: &'static str = "state";
+    const DB_PATH_SUFFIX: &'static str = "state";
+
+    /// Create a new [`FlatStateDb`] from a path.
+    pub fn new(path: std::path::PathBuf) -> anyhow::Result<Self> {
+        let mut columns = vec![default_cf_descriptor(StateRootHashes::table_name())];
+        VersionedDB::<NomtStateValues<UserNamespace>>::add_column_families(&mut columns)?;
+        VersionedDB::<NomtStateValues<KernelNamespace>>::add_column_families(&mut columns)?;
+        let other =
+            Self::get_rockbound_options(columns).setup_db_in_path_with_column_descriptors(path)?;
+        let other = Arc::new(other);
+        let user = VersionedDB::<NomtStateValues<UserNamespace>>::from_db(other.clone())?;
+        let kernel = VersionedDB::<NomtStateValues<KernelNamespace>>::from_db(other.clone())?;
+        Ok(Self {
+            user,
+            kernel,
+            other,
+        })
+    }
+
+    #[allow(dead_code)]
+    /// Get the underlying [`rockbound::DB`] for the historical state. Used for testing only.
+    pub fn get_db(&self) -> Arc<rockbound::DB> {
+        self.other.clone()
+    }
+
+    /// Get the underlying [`VersionedDB`] for the user state.
+    pub fn get_user_db(&self) -> &VersionedDB<NomtStateValues<UserNamespace>> {
```

Comment:
Do we want to keep these methods completely public instead of pub(crate)?

Asking because it is often gets abused and it is hard to backtrack on that.

Reply:
I don't have strong pushback on this one. Wanted to make sure it is a deliberate decision